users/marketplace/views.py

```diff
@@ -72,3 +72,48 @@ def view(request, listing_id):
     context["firstname"] = request.user.first_name
 
     return render(request, "marketplace/view_listing.html", context)
+
+
+@login_required
+def my_listings(request):
+    listings = Listing.objects.filter(owner=request.user)
+
+    context = {"listings": listings}
+    context["firstname"] = request.user.first_name
+
+    return render(request, "marketplace/my_listings.html", context)
+
+
+@login_required
+def update_listing(request, listing_id):
+    listing = Listing.objects.get(id=listing_id)
+
+    if request.method == "POST":
+        neighborhood = Neighborhood.objects.get(pk=request.POST["neighborhood"])
+        listing.title = request.POST["title"]
+        listing.description = request.POST["description"]
+        listing.price = request.POST["price"]
+        listing.email = request.POST["email"]
+        listing.phone = request.POST["phone"]
+        listing.address = request.POST["address"]
+        listing.neighborhood = neighborhood
+        listing.save()
+
+        messages.success(request, "Listing updated successfully")
+
+        id = listing.id
+        return HttpResponseRedirect(reverse("user_listings"))
+
+    neighborhoods = Neighborhood.objects.all()
+    context = {"listing": listing, "neighborhoods": neighborhoods}
+    context["firstname"] = request.user.first_name
+
+    return render(request, "marketplace/update_listing.html", context)
+
+
+@login_required
+def delete_listing(request, listing_id):
+    listing = Listing.objects.get(id=listing_id)
+    listing.delete()
```

Comment:
Do you think we should ask for any confirmation before deleting it?

We can create a new template `marketplace/delete_listing.html` and check the confirmation here.

WDYT?